R/schema_NamedRange.R

```diff
@@ -0,0 +1,30 @@
+# https://developers.google.com/sheets/api/reference/rest/v4/spreadsheets#NamedRange
+new_NamedRange <- function(namedRangeId,
+                           name,
+                           range) {
+  x <- list(
+    namedRangeId = namedRangeId,
+    name = name,
+    range = range
+  )
+  structure(
+    validate_NamedRange(x),
+    class = "NamedRange"
+  )
+}
+
+validate_NamedRange <- function(x) {
+  # I think read-only vs. required vs. optional status of these elements
+  # depends on what you're trying to do
+  maybe_string(x$namedRangeId, "namedRangeId")
+  maybe_string(x$name, "name")
+
+  # range is an instance of GridRange
```

Comment:
The schema often refer to yet more schema, some of which I haven't implemented. In which case I use this sort of comment in the validator.

Reply:
Although in this case, I have implemented `GridRange`, so I should fix this! But you get the idea.